src/corelibs/U2Gui/src/options_panel/OPWidgetFactory.h

```diff
@@ -69,6 +71,8 @@ class U2GUI_EXPORT OPFactoryFilterVisitorInterface {
     virtual bool typePass(ObjectViewType factoryViewType) = 0;
     virtual bool alphabetPass(DNAAlphabetType factoryAlphabetType) = 0;
     virtual bool atLeastOneAlphabetPass(DNAAlphabetType factoryAlphabetType) = 0;
+    // Returns true, if there is a DNA alphabet (default or extended).
+    virtual bool atLeastOneDnaPass() const = 0;
```

Comment:
Please see behavior for _common_data/fasta/RNA.fa: there is no PCR Primer Design tab, but there is In Silico PCR tab which is useless for current sequences

Reply:
Could you please provide a details how "atLeastOneAlphabetPass" is different from "atLeastOneDnaPass" in case if you want to match DNA sequences only? I think "atLeastOneAlphabetPass" will do the job

Reply:
`atLeastOneAlphabetPass` approves files that contain one or more sequences with an alphabet defined as `DNAAlphabetType`.
`atLeastOneDnaPass` approves files that contain one or more sequences with an alphabet identified as DNA or extended DNA.

Reply:
I see now: DNAAlphabetType does not differentiate between RNA and DNA. You are right. In this case I would use a serializable alphabet-id as the filer, not a runtime pointer. You can rework it as suggested or resolve the issue in the current state.

Reply:
@yalgaer, please see the [changes](https://github.com/ugeneunipro/ugene/pull/619/commits/55c0760e9e073fe881410001c15049e817d1a613): is this what you wrote about?

Reply:
Yes, thank you.